### ISSUE TITLE
How to configure PHP?

### ISSUE THREAD

**Issue by jimmyeatsquaids** [wallacebrf/synology_SMART_disk_data_logger#1]
Hi,

I'm trying to figure out how to configure PHP ony Synology DS418play for this to work. I'm currently getting stuck when trying to enter the values in the form (smart_config.php) as I can't successfully hit the "Submit" button to have the info saved. I would really appreciate some help in understanding how I need to setup PHP on my Synology to get this done.

**Comment by wallacebrf** [comment]
hello,

first, i suggest using this newer code here:
https://github.com/wallacebrf/SMART-to-InfluxDB-Logger

so to start, you do have PHO installed, and you are able to actually see the configuration screen, you are only not able to hit the submit button? 

what happens when you hit the submit button? does the page re-load and you simply do not see the parameters change? do you see any errors or other notices? 

did you enable the read/write permissions as required?

**Comment by jimmyeatsquaids** [comment]
Thanks for the fast reply!

I accidentally wrote on the wrong github page it seems. I'm aware of the new code and followed the instructions there. I have PHP installed and tried to configure Web Station but I'm not 100% sure I did it right. I took care of the read/write permissions as instructed and I can see the config screen:
![image](https://github.com/user-attachments/assets/be6504b8-5b66-47d8-b0ca-509e593a6a27)

After entering the values and hitting the Submit button I get error code 500:
![image](https://github.com/user-attachments/assets/fed09c6b-6a6e-41f2-900b-c708bfdd0232)

Sorry for writing on the wrong github page... but I really appreciate the help!

**Comment by wallacebrf** [comment]
what value do you have this line inside ```SMART_server2_config.php``` set to?

```$form_submittal_destination="index.php?page=6&config_page=smart_server2"; //set to the destination the HTML form submit should be directed to```

it looks like you have the PHP file loaded directly, and not being included within another file.

the screen that shows the configuration options, can you show me the URL in the address bar? you can block out the IP/domain, i just need to see what the path is, as you PROBABLY just need to change the value to 

```$form_submittal_destination="SMART_server2_config.php";```

**Comment by jimmyeatsquaids** [comment]
Yes I have the file loaded directly, the URL is: "https://192.168.178.252/config/smart_config.php"

The line for the form submittal was set to: `$form_submittal_destination="smart_config.php";`

Please note again I'm using the files and instructions from your newer github page so there's no "SMART_server2_config.php"

**Comment by wallacebrf** [comment]
change the line to 
```$form_submittal_destination="/config/smart_config.php";```

**Comment by jimmyeatsquaids** [comment]
I originally had it that way but changed it as both give the same result it seems. The URL my browser shows is also the same. I just tested again with the added /config/ in the path but it's the same error code 500. I feel like I probably did something wrong when settings up the web station as I'm not familiar with it at all.

**Comment by jimmyeatsquaids** [comment]
So I've been trying all sorts of things with web station and quadruple checked permissions but I can't find any issues on my end.

One thing I noticed when going through the files is some inconsistencies for the config file. In smart_config.php the path for the config file is NOT the same as in smart.sh:

smart.sh: `config_file_location="/volume1/web/config"	` `config_file_name="smart_logging_config.txt"`
smart_config.php: `$config_file="/volume1/Server_NVR/web/logging/smart_logging_config.txt";`

I changed this on my end as it is stated in the docs that these 2 need to line up but I it still doesn't work.

At this point I'm not quite sure what else I can try.

**Comment by jimmyeatsquaids** [comment]
Update: I finally got it to work and as predicted it was a problem with how I set up web station. I needed to configure an https port (http port was not enough) and access the smart_config.php through that port and then the "Submit" button finally worked.

I successfully got the smart data in the influxdb database but I'm currently in the process of upgrading to higher capacity drives so I will wait before configuring the JSON file for Grafana until that is done.
